File: 01_Language/01_Functions/php/explode.php
<?php

// Example 1
$pizza  = "piece1 piece2 piece3 piece4 piece5 piece6";
$pieces = explode(" ", $pizza);
echo $pieces[0] . "\n"; // piece1
echo $pieces[1] . "\n"; // piece2

// Example 2
$data = "foo:*:1023:1000::/home/foo:/bin/sh";
list($user, $pass, $uid, $gid, $gecos, $home, $shell) = explode(":", $data);
echo $user . "\n"; // foo
echo $pass . "\n"; // *

$str = 'one|two|three|four';

// positive limit
print_r(explode('|', $str, 2));

// negative limit (since PHP 5.1)
print_r(explode('|', $str, -1));
print_r(explode('|', $str, 0));

File: 01_Language/01_Functions/python/explode.py
# coding: utf-8

def explode(delimiter, string, limit=None):
    if limit is None:
        return string.split(delimiter)
    elif limit < 0:
        return string.split(delimiter)[:limit]
    elif limit == 0:
        return [string]
    else:
        return string.split(delimiter, limit - 1)


if __name__ == '__main__':
    pizza = "piece1 piece2 piece3 piece4 piece5 piece6"
    print(explode(" ", pizza))

    data = "foo:*:1023:1000::/home/foo:/bin/sh"
    print(explode(":", data))

    s = "one|two|three|four"
    print(explode("|", s, 2))
    print(explode("|", s, -1))
    print(explode("|", s, 0))
